src/level/level_loader.h

```diff
@@ -0,0 +1,158 @@
+/*
+ * This file is part of the Colobot: Gold Edition source code
+ * Copyright (C) 2001-2016, Daniel Roux, EPSITEC SA & TerranovaTeam
+ * http://epsitec.ch; http://colobot.info; http://github.com/colobot
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.
+ * See the GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program. If not, see http://gnu.org/licenses
+ */
+
+/**
+ * \file level/level_loader.h
+ */
+
+#pragma once
+
+#include "level/parser/parser.h"
+
+#include <string>
+
+/**
+ * \brief List of modes used in CreateScene()
+ */
+enum LevelLoadMode
+{
+    Normal          = 1, //!< used when starting new game
+    Reset           = 2, //!< used when restarting level
+    LoadSavedGame   = 4  //!< used when loading saved game
+};
+
+typedef std::function<void(CLevelLoader*, CLevelParserLine*, LevelLoadMode)> CommandFunction;
+
+/**
+ * \brief Tool to loading and building scene from file
+ */
+class CLevelLoader
+{
+public:
+    /**
+     * \brief Constructor, should not be used from outside CLevelLoader class,
+     * use CreateScene() instead.
+     */
+    CLevelLoader(bool solutionEnabled);
```

Comment:
You can probably make the constructor private if you don't want it to be invoked manually